packages/roosterjs-editor-core/lib/coreApi/getSelectionRangeEx.ts

```diff
@@ -0,0 +1,84 @@
+import { contains, createRange, safeInstanceOf, VTable } from 'roosterjs-editor-dom';
+import {
+    EditorCore,
+    GetSelectionRangeEx,
+    SelectionRangeEx,
+    SelectionRangeTypes,
+} from 'roosterjs-editor-types';
+
+const TABLE_SELECTED = '_tableSelected';
+
+/**
+ * @internal
+ * Get current or cached selection range
+ * @param core The EditorCore object
+ * @returns A Range object of the selection range
+ */
+export const getSelectionRangeEx: GetSelectionRangeEx = (core: EditorCore) => {
+    let result: SelectionRangeEx = null;
+    if (core.lifecycle.shadowEditFragment) {
+        const shadowRange =
+            core.lifecycle.shadowEditSelectionPath &&
+            createRange(
+                core.contentDiv,
+                core.lifecycle.shadowEditSelectionPath.start,
+                core.lifecycle.shadowEditSelectionPath.end
+            );
+
+        if (shadowRange.collapsed) {
+            const tableSelected = getTableSelected(core.contentDiv);
+
+            if (tableSelected) {
+                return createTableSelectionEx(tableSelected);
+            }
```

Comment:
Let's just remove this for now. We can go directly with better approach when we want to fix it.

Take some note that we have this pending work to fix